pkg/segment/structs/evaluationstructs.go

```diff
@@ -357,6 +357,57 @@ func (self *SortExpr) ReleaseProcessedSegmentsLock() {
 	self.processedSegmentsLock.Unlock()
 }
 
+func findNullFields(fields []string, fieldToValue map[string]utils.CValueEnclosure) ([]string, error) {
+	nullFields := []string{}
+	for _, field := range fields {
+		val, exists := fieldToValue[field]
+		if !exists {
+			return []string{}, fmt.Errorf("findNullFields: Expression has a field for which value is not present")
+		}
+		if val.Dtype == utils.SS_DT_BACKFILL {
+			nullFields = append(nullFields, field)
+		}
```

Comment:
Why do you return an error on line 365 instead of just adding the field to the list of `nullFields`?

Reply:
Actually !exists is different from NULL field. !exists mean that field was present in the query but we did not fetched the value for it. Even if it is NULL it would still show up as exists. In most of the methods like isnull etc. We are using this check and returning error cause the fieldToValue should have this entry but it does not have it right now so it is an error.